cou/cli.py

```diff
@@ -42,110 +71,333 @@ def parse_args(args: Any) -> argparse.Namespace:
+    )
+    plan_subparser.add_parser(
+        "control-plane",
+        description="Show the steps for upgrading the control-plane components.",
+        help="Show the steps for upgrading the control-plane components.",
+        usage="cou plan control-plane [options]",
+        parents=[subcommand_common_opts_parser],
+        formatter_class=CapitalisedHelpFormatter,
+    )
+
+    plan_subparser.add_parser(
+        "data-plane",
+        description="Show the steps for upgrading the data-plane components.",
+        help="Show the steps for upgrading the data-plane components.",
+        usage="cou plan data-plane [options]",
+        parents=[subcommand_common_opts_parser, dp_subparser],
+        formatter_class=CapitalisedHelpFormatter,
+    )
+
+    # Arg parser for "cou run" sub-command and set up common options
+    run_args_parser = argparse.ArgumentParser(add_help=False)
+    run_args_parser.add_argument(
         "--interactive",
         help="Run upgrade with prompt.",
         action=argparse.BooleanOptionalAction,
         default=True,
     )
-    parser.add_argument(
-        "-h",
-        "--help",
-        action="help",
-        default=argparse.SUPPRESS,
-        help="Show this help message and exit.",
+    run_args_parser.add_argument(
+        "--i-really-mean-it",
+        help="Run upgrade steps even if they are considered as risky operations (e.g. upgrading "
+        "data-plane nodes with non-empty computes).",
+        action="store_true",
+        default=False,
+    )
+    run_parser = subparsers.add_parser(
+        "run",
+        description="Run the cloud upgrade.",
+        help="Run the cloud upgrade.",
+        usage="cou run [options]",
+        parents=[subcommand_common_opts_parser, run_args_parser],
+        formatter_class=CapitalisedHelpFormatter,
     )
 
-    return parser.parse_args(args)
+    # Create control-plane and data-plane child commands to run partial upgrades
+    run_subparser = run_parser.add_subparsers(
+        title="Upgrade group",
+        dest="upgrade-group",
+        help="For more information about a upgrade group, run 'cou run <upgrade-group> -h'.",
+    )
+    run_subparser.add_parser(
+        "control-plane",
+        description="Run upgrade for the control-plane components.",
+        help="Run upgrade for the control-plane components.",
+        usage="cou plan control-plane [options]",
+        parents=[subcommand_common_opts_parser, run_args_parser],
+        formatter_class=CapitalisedHelpFormatter,
+    )
+    run_subparser.add_parser(
+        "data-plane",
+        description="Run upgrade for the data-plane components.",
+        help="Run upgrade for the data-plane components.",
+        usage="cou plan data-plane [options]",
+        parents=[
+            subcommand_common_opts_parser,
+            dp_subparser,
+            run_args_parser,
+        ],
+        formatter_class=CapitalisedHelpFormatter,
+    )
```

Comment:
we should mention that the upgrade will start only if the control plane has already been upgraded